Wrapping/Generators/Python/PyBase/pyBase.i

```diff
@@ -450,6 +450,63 @@ str = str
                     self.__SetDirection_orig__(matrix)
                 else:
                     self.__SetDirection_orig__(direction)
+
+            def keys(self):
+                """Return keys related to the image's metadata.
+
+                These keys are used in the dictionary resulting from dict(image).
+
+                These keys include MetaDataDictionary keys along with
+                'origin', 'spacing', and 'direction' keys, which
+                correspond to the image's Origin, Spacing, and Direction. However,
+                they are in (z, y, x) order as opposed to (x, y, z) order to
+                correspond to the indexing of the shape of the pixel buffer
+                array resulting from np.array(image).
+                """
+                meta_keys = self.GetMetaDataDictionary().GetKeys()
+                # Ignore deprecated, legacy members that cause issues
+                result = list(filter(lambda k: not k.startswith('ITK_original'), meta_keys))
+                result.extend(['origin', 'spacing', 'direction'])
```

Comment:
Perhaps assert that `origin`, `spacing`, and `direction` are not supplied by `meta_keys = self.GetMetaDataDictionary().GetKeys()`.  If they were, their values there would get hidden, which would be hard to debug.

Reply:
Worth considering, but for simplicity, `origin`, `spacing`, `direction` keys also in the MetaDataDictionary should not be a valid, supported state. We do not want duplication, and the `Origin`, `Spacing`, and `Direction` ivars should be used for these.